pegasus/helpers/section_api_helpers.rb

```diff
@@ -63,13 +60,49 @@ def self.fetch_if_allowed(id_or_ids, dashboard_user_id)
       server(:default).
       first
 
-    if row.nil?
-      return
-    end
+    return if row.nil?
 
     row.merge(age: birthday_to_age(row[:birthday]))
   end
 
+  # @param ids [Array[Integer]] the IDs to fetch.
+  # @param dashboard_user_id [Integer] the ID of the user doing the fetching.
+  # @returns [Array[Hash | nil]] an array, one entry per requested ID, with each
+  #   entry being a hash (representing the requested user) or nil (if the
+  #   requested user does not exist or is accessible by dashboard_user_id).
+  def self.fetch_if_allowed_array(ids, dashboard_user_id)
+    user = Dashboard::User.get(dashboard_user_id)
+    return ids.map {|_id| nil} unless user
+
+    allowed_ids = user.admin? ? ids : user.get_followed_bys(ids)
+    allowed_rows = Dashboard.db[:users].
+      where(users__id: allowed_ids, users__deleted_at: nil).
+      left_outer_join(:secret_pictures, id: :secret_picture_id).
+      select(*fields,
+        :users__birthday___birthday,
+        :secret_pictures__name___secret_picture_name,
+        :secret_pictures__path___secret_picture_path
+      )
+
+    # Convert allowed_rows from an array of hashes (each representing a user)
+    # to a hash of hashes (keys of user_id, values representing a user).
+    allowed_rows = {}.tap do |allowed_rows_hash|
+      allowed_rows.each do |allowed_row|
+        allowed_rows_hash[allowed_row[:id]] = allowed_row
+      end
+    end
+
+    # Add user age to the hash.
+    ids.map do |id|
+      if allowed_rows.key? id
+        allowed_rows[id][:age] = birthday_to_age(allowed_rows[id][:birthday])
+      end
+    end
```

Comment:
Unfortunately, this isn't as clean and simple as I feel it should be. That said, I'm unsure how to simplify it as none of the variants of `select` seem to be helpful.